File: treatment/src/test/java/com/hartwig/actin/treatment/database/TrialConfigDatabaseUtilTest.java
package com.hartwig.actin.treatment.database;

import static org.junit.Assert.assertEquals;

import org.apache.logging.log4j.util.Strings;
import org.junit.Test;

public class TrialConfigDatabaseUtilTest {

    @Test
    public void canConvertToCohorts() {
        assertEquals(0, TrialConfigDatabaseUtil.toCohorts("all").size());
        assertEquals(1, TrialConfigDatabaseUtil.toCohorts("A").size());
        assertEquals(2, TrialConfigDatabaseUtil.toCohorts("A;B").size());
    }

    @Test(expected = IllegalArgumentException.class)
    public void crashOnMissingCohortParam() {
        TrialConfigDatabaseUtil.toCohorts(Strings.EMPTY);
    }

    @Test
    public void canConvertToParams() {
        assertEquals(0, TrialConfigDatabaseUtil.toParameters(Strings.EMPTY).size());
        assertEquals(1, TrialConfigDatabaseUtil.toParameters("param1").size());
        assertEquals(2, TrialConfigDatabaseUtil.toParameters("param1;4").size());
    }
}

File: treatment/src/test/kotlin/com/hartwig/actin/trial/trial/TrialConfigDatabaseUtilTest.kt
package com.hartwig.actin.trial.trial

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class TrialConfigDatabaseUtilTest {

    @Test
    fun canConvertToReferenceIds() {
        assertThat(TrialConfigDatabaseUtil.toReferenceIds("all")).hasSize(1)
        assertThat(TrialConfigDatabaseUtil.toReferenceIds("I-01")).hasSize(1)
        assertThat(TrialConfigDatabaseUtil.toReferenceIds("")).isEmpty()

        val referenceIds = TrialConfigDatabaseUtil.toReferenceIds("I-01, I-02")
        assertThat(referenceIds).containsExactlyInAnyOrder("I-01", "I-02")
    }

    @Test
    fun canConvertToCohorts() {
        assertThat(TrialConfigDatabaseUtil.toCohorts("all")).hasSize(0)
        assertThat(TrialConfigDatabaseUtil.toCohorts("A")).hasSize(1)

        val cohorts = TrialConfigDatabaseUtil.toCohorts("A, B")
        assertThat(cohorts).containsExactlyInAnyOrder("A", "B")
    }

    @Test(expected = IllegalArgumentException::class)
    fun crashOnMissingCohortParam() {
        TrialConfigDatabaseUtil.toCohorts("")
    }
}